### ISSUE TITLE
Incomplete argument documentation

### ISSUE THREAD

**Issue by hardianlawi** [VowpalWabbit/vowpal_wabbit#2229]
#### Describe the bug
There are missing explanations of the arguments on this page https://github.com/VowpalWabbit/vowpal_wabbit/wiki/Command-line-arguments

One example is `--cb_type`. What are the available options? Also, since there are so many arguments, what are the default behaviors if the arguments are not specified? e.g. whether the label is multiclass or binary, etc.

**Comment by jackgerrits** [comment]
That wiki page is definitely out of date I'll take this and get it updated soon.

[This tutorial](https://vowpalwabbit.org/tutorials/contextual_bandits.html#vowpal-wabbit-tutorial) on the website touches on `cb_type`, and includes references.

**Comment by hardianlawi** [comment]
@jackgerrits if I specify the `cb_type` to `ips`, what is the algorithm used as the learner? I have been searching for it, but I could not find any reference. My guess is linear models trained in One-vs-all manner.

**Comment by jackgerrits** [comment]
It will use cost sensitive one against all and that will in turn reduce to gradient descent by default. You can override the default by providing command line options. I don't think there is a good central place to see a list of base learners other than where the reduction stack is defined [here](https://github.com/VowpalWabbit/vowpal_wabbit/blob/8ccb5ed724a066ba6c30838d80eb825b8c8eacb3/vowpalwabbit/parse_args.cc#L1230).

**Comment by hardianlawi** [comment]
Thanks for your reply! Although it would be really helpful if you have some references that are more complete than the wiki.

For example, now I know they are not training `K` binary classifiers based on this [page](https://github.com/VowpalWabbit/vowpal_wabbit/wiki/Cost-Sensitive-One-Against-All-(csoaa)-multi-class-example)

> The reduction with --csoaa is to a regression problem (i.e. conditional mean estimation), so forcing the loss function to logistic does not make much sense. Generally, when using multi-class, you should leave the --loss_function alone and let the algorithm use the built-in default.

However, I do not know what loss function they are using and also the base algorithm as the learners. I will try to read the codes, but I am not really familiar with C so that would probably be my last option.

Based on this [page](http://users.umiacs.umd.edu/~hal/tmp/multiclassVW.html), I assume it is using squared loss to train the regressors. 

> Be careful when changing the loss function. If you're doing one of the regression-based models (csoaa) you need to use a regrssion loss and should probably stick to squared loss. If you're doing one of the classification-based models (oaa, wap, csoaa_ldf mc or wap_ldf) then you might want to use a classification loss, like logistic or hinge.

Also, when you said default gradient descent, were you referring to the gradient descent with the `--adaptive`, `--normalized` and `--invariant` flags on or just vanilla sgd?

I also found that it is using `--csoaa` by default from [here](https://github.com/VowpalWabbit/vowpal_wabbit/wiki/v7.0_tutorial.pdf)

**Comment by jackgerrits** [comment]
I agree, the coverage of the wiki is spotty. Any contributions to help improve the wiki are greatly appreciated. Yes, squared loss is the default as far as I can tell.

The GD reduction by default has all of the flags you mentioned turned off. See here:
https://github.com/VowpalWabbit/vowpal_wabbit/blob/8ccb5ed724a066ba6c30838d80eb825b8c8eacb3/vowpalwabbit/gd.cc#L1120

**Comment by JohnLangford** [comment]
Closing as addressed. The specific example raised here (`--cb_type` options) is now documented in the wiki's [Contextual Bandit algorithms](https://github.com/VowpalWabbit/vowpal_wabbit/wiki/Contextual-Bandit-algorithms) page and [Command line arguments](https://github.com/VowpalWabbit/vowpal_wabbit/wiki/Command-Line-Arguments) page. Many other option help strings have also been improved in the codebase (visible via `--help`).
